sdks/java/core/src/test/java/org/apache/beam/sdk/io/DefaultFilenamePolicyTest.java

```diff
@@ -32,66 +32,89 @@
 
   @Test
   public void testConstructName() {
-    assertEquals("output-001-of-123.txt",
-        constructName("output", "-SSS-of-NNN", ".txt", 1, 123, null, null));
+    assertEquals(FileSystems.matchNewResource("path/to/output-001-of-123.txt", false),
```

Comment:
On first glance, the helper would have the same params, so would not help that much.